### ISSUE TITLE
Anyone can become admin

### ISSUE THREAD

**Issue by scoopgracie** [apertium/phenny#534]
This is a pretty major security bug. Anyone can become admin if one of the admins if offline.

`/nick <admin-name>`; `/msg begiak .adminonlycommand`; `/nick <normal-name>`

**Comment by sushain97** [comment]
That's why admins should have registered nicks.

**Comment by scoopgracie** [comment]
Yeah, but that doesn't stop you from doing something in the short time before NickServ renicks you.

**Comment by scoopgracie** [comment]
Freenode requires SASL login; why not do adminship by SASL name instead of nick? SASL is much harder to spoof.

**Comment by scoopgracie** [comment]
I still think this needs a fix.

**Comment by scoopgracie** [comment]
There has been some discussion on IRC, and we need to discuss solutions. We need to 
1. Do something with NickServ to add protection,
2. Create a dedicated admin nick and not use it in public channels to keep it a secret, or
3. Use a different mechanism altogether.

**Comment by wei2912** [comment]
How would awarding admin permissions based on SASL authentication work? As
I understand it's merely used to bypass the need to send NickServ the
password in plaintext, and doesn't change the fact that somebody could
still take control of the nick for half a minute.

I'd recommend only allowing admin if the person has been verified by
NickServ. This would require an additional command to NickServ.

**Comment by scoopgracie** [comment]
Oh, okay. I'd misunderstood sasl. I think NickServ verification is a good idea.

**Comment by scoopgracie** [comment]
Another thought is to replace all admin commands with a Web panel.

**Comment by scoopgracie** [comment]
Or, only let admin commands work in a special channel in which only admins are invited. This would have the added benefit of letting admins other than those with server access grant/revoke adminship.
